{
	var a = 1;
	a := 1 / a;
	
	write("-> trycatch");
	
	(try
		a := 0;
		write(a);
		a := 2 / a;
		write("passou do erro")
	catch(msgErro)
		if msgErro=="divisaoPorZero" 
		then write("denominador nao poder ser zero")
		else write("erro desconhecido"));
	
	write("-> throw com string");
	
	(try
		a := 5;
		(if a==5	then
			throw("throw str")
		else
			write("sem erro"));
		write("passou do erro")
	catch(msgErro)
		write("erro foi: "++msgErro));
	
	write("-> throw com inteiro");
	
	(try
		throw(999)
	catch(e)
		if e==999 then
			e := e + 1;
			write(e)
		else
			write("outro erro"));
	
	write("-> throw com boolean");
	
	(try
		a := 0;
		throw(a==1)
	catch(e)
		if e then
			write("throw then")
		else
			write("throw false"));
	
	write("fim")
}
